lib/plugins/digging.js

```diff
@@ -7,12 +7,23 @@ module.exports = inject
 
 function inject (bot) {
   let swingInterval = null
-  let waitTimeout = null
+  let waitInterval = null
 
   let diggingTask = createDoneTask()
 
   bot.targetDigBlock = null
   bot.lastDigTime = null
+  bot.digPercentage = 0
+  bot.lastHeldItemName = ''
+  if (bot.heldItem) { bot.lastHeldItemName = bot.heldItem.name }
+  bot.on('heldItemChanged', () => {
+    let nameToCompare = ''
+    if (bot.heldItem) { nameToCompare = bot.heldItem.name }
+    if (bot.lastHeldItemName !== nameToCompare) {
```

Comment:
I think any held item change cancels the dig process? Even increasing the item amount in a slot.

Reply:
Surprisingly no. If you have two pickaxes with equal durability and/or nbt, and switch between them, or seitch between different fist slots while mining, it won't cancel the mining timer. Mining a tree with fists and picking up a log while mining will reset it, though constantly mining with logs while the stack size increases doesn't last time I checked. I'll double check later today but I'm pretty sure this is correct.

Though now that I'm looking at this the system here won't take into account durability/enchant differences, though that's just nbt and can likely be changed easily. This system will still work in most cases though and is better than nothing, but it should be fixed.

Reply:
A better way to check if items are equal is the stringify the item and compare that. You could make a string of the item you started mining with and reset the progress if the string of the held item changes.